hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/factory/HoodieSparkKeyGeneratorFactory.java

```diff
@@ -76,7 +76,7 @@ public class HoodieSparkKeyGeneratorFactory {
 
   public static KeyGenerator createKeyGenerator(TypedProperties props) throws IOException {
     String keyGeneratorClass = getKeyGeneratorClassName(props);
-    boolean autoRecordKeyGen = KeyGenUtils.enableAutoGenerateRecordKeys(props);
+    boolean autoRecordKeyGen = KeyGenUtils.enableAutoGenerateRecordKeys(props) && !props.getBoolean("hoodie.internal.sql.merge.into.writes",false);
```

Comment:
Fixed. In SQL keygen we update this anyway, so previously wasn't that unnecessary, because it would get overwritten? Now we need it for this, but before I don't think sqlkeygen was being used at all for pkless.

Reply:
I don't see any docs ?